ec2-agents.tf

```diff
@@ -1,149 +0,0 @@
-## Identity for Jenkins Controller infra.ci.jenkins.io
-data "aws_iam_user" "jenkins_infra_ci" {
-  user_name = "jenkins-infra-ci"
-}
-
-resource "aws_iam_policy" "jenkins_ec2_agents" {
-  # IAM Policy from https://plugins.jenkins.io/ec2/#user-content-iam-setup
-  name        = "jenkins_ec2_agents"
-  path        = "/"
-  description = "IAM Policy to allow a Jenkins Controller to start and manage EC2 agents with the 'ec2' plugin."
-
-  policy = data.aws_iam_policy_document.jenkins_ec2_agents.json
-}
-
-## Allow wildcard for resource as the EC2 instance IDs are not known in advance
-#trivy:ignore:AVD-AWS-0342 trivy:ignore:aws-iam-no-policy-wildcards
-data "aws_iam_policy_document" "jenkins_ec2_agents" {
-  statement {
-    sid    = "Stmt1312295543082"
-    effect = "Allow"
-
-    actions = [
-      "ec2:DescribeSpotInstanceRequests",
-      "ec2:DescribeSpotPriceHistory",
-      "ec2:CancelSpotInstanceRequests",
-      "ec2:GetConsoleOutput",
-      "ec2:RequestSpotInstances",
-      "ec2:RunInstances",
-      "ec2:StartInstances",
-      "ec2:StopInstances",
-      "ec2:TerminateInstances",
-      "ec2:CreateTags",
-      "ec2:DeleteTags",
-      "ec2:DescribeInstances",
-      "ec2:DescribeKeyPairs",
-      "ec2:DescribeRegions",
-      "ec2:DescribeImages",
-      "ec2:DescribeAvailabilityZones",
-      "ec2:DescribeSecurityGroups",
-      "ec2:DescribeSubnets",
-      "iam:ListInstanceProfilesForRole",
-      "iam:PassRole",
-      "ec2:GetPasswordData"
-    ]
-
-    resources = ["*"]
-  }
-}
-
-
-resource "aws_iam_user_policy_attachment" "allow_ec2_on_infraci" {
-  user       = data.aws_iam_user.jenkins_infra_ci.user_name
-  policy_arn = aws_iam_policy.jenkins_ec2_agents.arn
-}
-
-## Identity for Jenkins Controller ci.jenkins.io
-data "aws_iam_user" "jenkins_ci" {
-  user_name = "jenkins-ci"
-}
```

Comment:
Note: to be deleted manually (tracked in https://github.com/jenkins-infra/helpdesk/issues/3662#issuecomment-1770472278)